src/comment_parser_tests.rs

```diff
@@ -94,6 +94,17 @@ fn test_italic() {
     );
 }
 
+/// Two underscores in a row are treated as normal text.
```

Comment:
I'm not sure what the code in the main file does, but just to be sure, since this comment is not clear: two underscores in a row are transformed into a single underscore character, correct ?

Reply:
Actually `metamath-knife` parses comments in two steps:
- during the first step, it generates a "map" of the comment, noting where math mode, italics start and end, where subscript starts, etc. and the text spans between those. This only returns the _positions_ of the start/end, etc. No actual text processing is done.
- during the second step, the caller can call either an `unescape_text` method, only on the text parts, to get the transformed string.

So to properly handle double underscores, we need to modify both steps:
- in the first steps, we need to stop treating double underscores as subscripts as it was the case before. This is what this PR does. The test verifies that a single `Text` span is returned, instead of a first `Text`, a subscript marker, and a second `Text`.
- in the second step, we need to actually replace two underscore by one, at the same time we do the unescaping of other characters. This PR does not do this second part (yet?)

Reply:
Ok, I realize that the second step is actually really easy to modify, adding a commit now.